xboxdash/main.c

```diff
@@ -1,21 +1,61 @@
-#include <xboxrt/debug.h>
-#include <pbkit/pbkit.h>
+#include <winapi/windows.h>
+#include <hal/debug.h>
+#include <hal/video.h>
 #include <hal/xbox.h>
-#include "stdio.h"
+#include <xboxkrnl/xboxkrnl.h>
+
+ULONG status = 0;
+ULONG oldstatus = 0;
+void checkDrvStat();
+void dvdStat16();
+void dvdStat96();
+void txtWall();
 
 void main(void)
 {
-    int i;
-
-    if (pb_init()) {
-        XSleep(2000);
-        XReboot();
-        return;
+    XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+    debugPrint("Please insert a disc...\n");
+    for (;;)
+    {
+        checkDrvStat();
     }
+}
 
-    pb_show_debug_screen();
-    debugPrint("Please insert an Xbox disc...\n");
-    while(1) XSleep(1000);
-    pb_kill();
-    XReboot();
+void checkDrvStat(void) // thanks to the people helping me in the off-topic channel on the xemu discord server. here we have a very finicky and probably buggy implementation of a dvd autolauncher.
+{
+    HalReadSMCTrayState(&status, NULL);
+    switch (status)// using a switch now, I think this just looks cooler. if statements work fine I guess idk.
+    {
+    case 16: // if satus is 16, we know the tray is open.
```

Comment:
Consider using hexadecimal instead (0x10 and 0x60), looks cleaner than regular decimal.